mage_ai/data_preparation/models/pipeline.py

```diff
@@ -311,7 +308,7 @@ def update_block(self, block, upstream_block_uuids=None):
                     ]
                 block.upstream_blocks = self.get_blocks(upstream_block_uuids)
         self.blocks_by_uuid[block.uuid] = block
-        self.save()
+        self.save(block_uuid=block.uuid)
```

Comment:
this is a little bit tricky. we also update the downstream_blocks field in other blocks in this method.

Reply:
oh good point...

Reply:
ok i just updated it so that it'll save the entire pipeline if we're updating upstream blocks